### PR TITLE
Update TypeScript guide

### PR DESCRIPTION
**Pull request by ghiculescu**

### CODE REVIEW

**Comment by ghiculescu** [comment]
@marcoroth could you re-run CI? I don't think the errors are related.

File: docs/reference/using_typescript.md [hunk 1]
```diff
@@ -38,12 +36,12 @@ export default class MyController extends Controller {
 }
 ```
 
-> The `declare` avoids override of the exist stimulus property and just define the type for TypeScript.
+> The `declare` avoids overriding the exist Stimulus property, and just defines the type for TypeScript.
```
Comment (marcoroth) [inline docs/reference/using_typescript.md:39]: Should this be `existing`? Also might be worth to mention that `declare` is a keyword
```suggestion
> The `declare` keyword avoids overriding the existing Stimulus property, and just defines the type for TypeScript.
```

File: docs/reference/using_typescript.md [hunk 2]
```diff
@@ -57,11 +55,11 @@ export default class MyController extends Controller {
...
```
Comment (marcoroth) [inline docs/reference/using_typescript.md:58]: Same here
```suggestion
> The `declare` keyword avoids overriding the existing Stimulus property, and just defines the type for TypeScript.
```

File: docs/reference/using_typescript.md [hunk 3]
```diff
@@ -2,14 +2,12 @@
...
-stimulus properties.
+The following documentation shows how to define types for
+Stimulus properties.
 
 ## Define Controller Element Type
 
-By default, the `element` of the controller is from type `Element`.  
-If the element expected type is as example `HTMLFormElement` it can
-be defined this way:
+By default, the `element` of the controller is from type `Element`. You can override this in your `extends` statement. For example, if the element type is expected to be a `HTMLFormElement`:
```
Comment (marcoroth) [inline docs/reference/using_typescript.md:10]:
```suggestion
By default, the `element` of the controller is of type `Element`. You can override this in your `extends` statement. For example, if the element type is expected to be a `HTMLFormElement`:
```

File: docs/reference/using_typescript.md [hunk 4]
```diff
@@ -2,14 +2,12 @@
 
 Stimulus itself is written [TypeScript](https://www.typescriptlang.org/)
```
Comment (marcoroth) [inline docs/reference/using_typescript.md:3]: I think we are missing a `in` here

**Comment by marcoroth** [comment]
Also while we are at it: the code examples on this page use 4 spaces whereas all the other examples in the docs use 2. I think we should also use 2 here

**Comment by ghiculescu** [comment]
Thanks for double checking my changes @marcoroth !

**Review by marcoroth — APPROVED** [review]
Hey @ghiculescu, thanks for the updates! This looks good from my side! 

While going over it again I noticed that it wasn't 100% complete. I just committed the remaining changes to your branch. If you are fine with this we can merge it.

**Comment by ghiculescu** [comment]
Looks great.

**Pull request merged**